src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
     )
 }
 
-function statFile(file: vscode.Uri) {
-    return statSync(file.fsPath).size
+interface BucketQuickPickItem extends vscode.QuickPickItem {
+    bucket: S3.Bucket | undefined
 }
 
-async function refreshNode(node: S3BucketNode | S3FolderNode, commands: Commands): Promise<void> {
-    node.clearChildren()
-    return commands.execute('aws.refreshAwsExplorerNode', node)
+/**
+ * 
+ * @returns 
+ */
+async function promptUserForBucket(): Promise<S3.Bucket | undefined> {
+    let regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1" //TODO:: decide how to get the regionCode!!
+    const s3client = ext.toolkitClientBuilder.createS3Client(regionCode)
+    const window = Window.vscode()
+    const allBuckets: S3.Bucket[] = await s3client.listAllBuckets()
+    const s3Buckets = allBuckets.filter(bucket => {
+        return bucket && bucket.Name
+    }) as S3.Bucket[]
+
+    
+    const createNewBucket: BucketQuickPickItem = {
+        label: localize('AWS.command.s3.createBucket',"Create new bucket"),
+        bucket: undefined
+    }
+    const bucketItems: BucketQuickPickItem[] = s3Buckets.map(bucket => {
+        return { 
+            label: bucket.Name!,
+            bucket
+        }
+    })
+
+    const picker = createQuickPick({
+        options: {
+            canPickMany: false,
+            ignoreFocusOut: true,
+            title: localize('AWS.message.selectBucket','Select S3 bucket to upload to'),//localize('AWS.lambda.upload.title', 'Select Upload Type'),
+            step: 2,
+            totalSteps: 2,
+        },
+        items: [...bucketItems, createNewBucket],
+        buttons: [vscode.QuickInputButtons.Back]
+    })
+    const response = verifySinglePickerOutput(await promptUser({
+        picker: picker,
+        onDidTriggerButton: (button, resolve, reject) => {
+            if (button === vscode.QuickInputButtons.Back) {
+                resolve([{
+                    label: "back",
+                    bucket: undefined
+                }])
+            }
+
+        },
+    })) 
+
+    if (!response) {
+        return undefined
+    }
+
+    if (!response.bucket) {
+        if (response.label === 'back') {
+            uploadFilePaletteCommand()
```

Comment:
Is there a better way of controlling the flow than this? This function won't resolve until `uploadFilePaletteCommand` is run to completion. What happens in terms of returning from this function?

At the very least, `uploadFilePaletteCommand` is async, so this should be awaited.